navigation-extender-runtime/src/main/java/com/foursoft/xml/JaxbContextFactory.java

```diff
@@ -45,12 +45,12 @@ private JaxbContextFactory() {
         throw new UnsupportedOperationException("JAXBContextFactory shall not be instantiated (static class");
     }
 
-    public static synchronized JAXBContext initializeContext(final String packageName) throws JAXBException {
+    public static synchronized JAXBContext initializeContext(final String packageName, final ClassLoader classLoader) throws JAXBException {
         JAXBContext context = jaxbContextCache.get(packageName);
 
         // not implemented with computeIfAbsent because .newInstance throws JAXBException
         if (context == null) {
-            context = JAXBContext.newInstance(packageName);
+            context = JAXBContext.newInstance(packageName, classLoader);

```

Comment:
@faios  @RainerGanss wouldn't it be the same to just make the following:
```suggestion
            context = JAXBContext.newInstance(packageName, JAXContextFactory.class.getClassLoader());
```
and leave the method signature of `initializeContext` as it is?

Reply:
I thought about it myself.
Most likely it will be the same. Since there are only 2 uses of the method I decided with the 100% solution. 
Why this is the 100% solution? Because JAXB does not care if the factory is in the same ClassLoader. It is important that the classes which are send to Jaxb (e.g. the NamespacePrefixMapperImpl) is in the same ClassLoader.

But I can change that if you want to.